mtg_draft_ai/deckbuild.py

```diff
@@ -1,12 +1,50 @@
 """Deckbuilding functions."""
 
 from mtg_draft_ai import synergy
+import networkx as nx
 
 
-_NONLANDS_IN_DECK = 23
+_NONLANDS_IN_DECK_DEFAULT = 23
 
+_COLOR_COMBOS = ['WU', 'WB', 'WR', 'WG', 'UB', 'UR', 'UG', 'BR', 'BG', 'RG']
 
-def _centralities_build(card_pool_graph):
+def _comm_score(current_graph, comm):
+    build_with_comm = list(current_graph.nodes) + list(comm)
+    graph_with_comm = synergy.create_graph(build_with_comm, remove_isolated=False)
+    edges_added = len(graph_with_comm.edges) - len(current_graph.edges)
+    return edges_added / len(comm)
+
+
+def _num_nonlands(current_build):
+    return len([c for c in current_build if 'land' not in c.types])
+
+
+def _communities_build(card_pool_graph, target_playables):
+    if _num_nonlands(card_pool_graph.nodes) < target_playables:
+        raise DeckbuildError('Not enough cards')
```

Comment:
This error message is not descriptive enough. The issue is not that you don't have enough cards, the issue is that you do not have enough **nonland** cards. Also, if Python does not by default list the arguments of a method in its error messages, this seems like an appropriate time to do so manually, since this case is both unlikely and very severe.

Reply:
I can tweak the error message. I don't think we need to include the arguments for a few reasons:

1). This case is somewhat common, since remember that `best_two_color_synergy_build` tries every color combination for every pool, and most of the color combos won't have enough.

2). The change to `best_two_color_synergy_build` to aim for a lower # of playables if none of the combos have 23 means that at least one color combo will not hit this.

3). Since deckbuilding occurs when you click the autobuild button (or when running a script locally), it's pretty easy to get the pool either from the site or from a log file and repro any errors that do come up.

IMO the line you draw to balance between logging and code verbosity vs. post-fact debuggability is tool and context dependent and I don't think there's a lot of value in refining it here when there are no users aside from ourselves, inputs which cause unexpected exceptions are easily reproducable, and django running in development mode gives a lot of debugging info.